### DISCUSSION TITLE
Config AltGr to be something else

### DISCUSSION THREAD

**Discussion by KaffeMyers** [houmain/keymapper#20] [Q&A] [answered]
I have this issue trying to map AltGr to become the 'Ext' for the right hand side.

I've tried a few alternatives, but stumble upon weird behavior every time.

Does anyone know how I would go about this for it to work as expected, if at all possible?

**Comment by houmain** [comment]
Do you mean you want both `IntlBackslash` and `AltRight` to become an `Ext` key? Or `AltRight` instead of `IntlBackslash`? Simply changing to `Ext = AltRight` does not work as expected?

**Reply by KaffeMyers** [reply]
Thanks for taking the time to reply!

Yes, I want both the `IntlBackslash` and `AltRight` to be used as the `Ext`

As a side note: `Ext = AltRight` as the only `Ext` key leads to it being both the AltGr layer function AND the extension key simultaneously, making it act rather funky.
 
Currently trying this on Windows with `interception`.

**Reply by KaffeMyers** [reply]
...so, to make it super clear: just like `MetaLeft` and `MetaRight` both working as `Meta`, I want a `ExtLeft` and `ExtRight` to work as `Ext`, where `ExtLeft = IntlBackslash` and `ExtRight = AltRight`

**Reply by houmain** [reply]
Sounds reasonable. Unfortunately there is no way to define new logical keys (like `Meta`) in the configuration file yet. You would have to duplicate each mapping :cry:.
`AltGr` is a very special key, which causes some problems on Windows. I cannot remember the details ATM but my solution was to always immediately forward it (by implicitly prepending a `AltRight >> AltRight` to the configuration). This is even done on Linux and Windows when `interception` is used (which may not have issues without it). I need to investigate this and try to come up with a better solution.

**Reply by KaffeMyers** [reply]
Ok, then I know. I will fiddle with the config a bit more to see if I can get that effect with some "duplication" and what not.

Enjoying the software so far! Looking forward to following it's development!

**Reply by KaffeMyers** [reply]
I sadly do not find any way to get this working properly even with the input given here. I hope you find a good way to deal with this issue!
Anyway, thanks again for this. :)

**Reply by houmain** [reply]
With the last commit (bb7154f03ccb5e33e10865300c7792a3e6c32ac5) `AltRight` is no longer immediately forwarded.

**Reply by houmain** [reply, marked as answer]
And with the commit cfa5101cea45004b90153fd99677007a90dc002a it is possible to define your own logical keys like:
`Ext = IntlBackslash | AltRight`
Now it should be much easier to create your envisioned configuration.

**Reply by KaffeMyers** [reply]
Now I've tried it out and it is working like a dream! Device/OS hopping is a breaze with this. Thanks a bunch.

**Reply by KaffeMyers** [reply]
For good measure, this is what I ended up with
https://gitlab.com/-/snippets/2219583
Based on a normal Swedish keyboard layout

**Reply by houmain** [reply]
Very cool! How did you generate this nice .png?
FYI: You could start a block with `[modifier="!Boss"]` instead of prepending `!Boss` to all mappings, as can be seen in my [example configuration](https://github.com/houmain/keymapper/blob/main/keymapper.conf).

**Reply by KaffeMyers** [reply]
[keyboard-layout-editor.com](http://www.keyboard-layout-editor.com/)

Nice, config updated!
